docs: refer to the app as the Mend Renovate app (#23336)
renovatebot/renovate: docs/usage/self-hosted-configuration.md

You may need to set a `username` if you:

- use the Bitbucket platform, or

> - use the GitHub App with CLI (required) // DRAFT: Should I change this line?

Reviewer: ```suggestion
- use a self-hosted GitHub App with CLI (required)
```
